R/cre.R

```diff
@@ -176,58 +188,75 @@ cre <- function(y, z, X, method_params, hyper_params){
 
 
   # Inference ------------------------------------------------------------------
-
   logger::log_info("Starting CATE Inference")
-  #message("Conducting Inference Subsample Analysis")
-
-  # Estimate ITE ---------------------------------------------------------------
-  logger::log_info("Estimating ITE...")
-  ite_list_inf <- estimate_ite(y = y_inf, z = z_inf, X = X_inf,
-                   ite_method = getElement(method_params,"ite_method_inf"),
-                   is_y_binary = getElement(method_params,"is_y_binary"),
-                   include_ps = getElement(method_params,"include_ps_inf"),
-                   ps_method = getElement(method_params,"ps_method_inf"),
-                   oreg_method = getElement(method_params,"oreg_method_inf"),
-                   X_names = X_names,
-                   include_offset = getElement(method_params,"include_offset"),
-                   offset_name = getElement(method_params,"offset_name"),
-                   random_state = getElement(method_params, "random_state"))
-
-  ite_inf <- ite_list_inf[["ite"]]
-  ite_std_inf <- ite_list_inf[["ite_std"]]
-  sd_ite_inf <- ite_list_inf[["sd_ite"]]
-
-  # Generate rules matrix ------------------------------------------------------
+
+  # Estimate ITE
+  if (is.null(ite)) {
+    logger::log_info("Estimating ITE...")
+    ite_list_inf <- estimate_ite(y = y_inf, z = z_inf, X = X_inf,
+                                 ite_method = getElement(method_params,"ite_method_inf"),
+                                 is_y_binary = getElement(method_params,"is_y_binary"),
```

Comment:
Same here, and the following lines.

Reply:
only this line: `'%>%' <- magrittr::'%>%'` ?

Reply:
Same what?